rules/firestore.json

```diff
@@ -35,50 +36,188 @@ service cloud.firestore {
+            allow update: if userIsVerified() && !userIsCreator(resource.data) &&
+                             userIsChosenOpponent() &&
+                             preservesField('creator') &&
+                             preservesField('firstPlayer') &&
+                             preservesField('partType') &&
+                             preservesField('maximalMoveDuration') &&
+                             preservesField('totalPartDuration') &&
+                             (preservesField('partStatus') || acceptsConfig()) &&
+                             (preservesField('chosenOpponent') || removesOneSelfFromChosenOpponent());
+
+            // Any verified user can read a joiner
+            allow read: if userIsVerified();
+        }
+        function joinerHasNotStarted(joiner) {
+            return joiner.partStatus < 3;
+        }
+        match /joiners/{partId}/candidates/{userId} {
+            // Users can only add themselves to the candidate list if they are not the creator and if they use the right username
+            allow create: if userIsVerified() && userId == request.auth.uid &&
+                             request.auth.uid != creatorOf(partId).id &&
+                             request.resource.data.name == usernameOf(request.auth.uid) &&
+                             request.resource.data.id == request.auth.uid;
+            // Users can only delete themselves
+            allow delete: if userIsVerified() && userId == request.auth.uid;
+            // No updates can be made
+            allow update: if false;
+            // Any verified user can read the candidate list
+            allow read: if userIsVerified();
+        }
+        function creatorOf(partId) {
+            return get(/databases/$(database)/documents/joiners/$(partId)).data.creator;
+        }
+        function userIsVerified() {
+            return request != null &&
+                   request.auth != null &&
                    request.auth.uid != null &&
                    request.auth.token.email_verified == true;
         }
-        function isAuthentified() {
-            return request != null &&
-                   request.auth != null;
+        function userIsCreator(joiner) {
+            return joiner.creator != null &&
+                   joiner.creator.id == request.auth.uid;
         }
-        function isOwner(userId) {
-            return request != null &&
-                request.auth != null &&
-                request.auth.uid == userId;
+        function userIsCreatorWithValidName(joiner) {
+            return userIsCreator(joiner) &&
+                   joiner.creator.name == usernameOf(request.auth.uid);
+        }
+        function preservesField(field) {
+            // A field is preserved if it is not part of the update
+            // or if the same value is given in the update
+            return !(field in request.resource.data) || (request.resource.data[field] == resource.data[field]);
```

Comment:
Y avais pas des updates ou request.resource.data contient field mais qu'il n'as pas pour autant changé ?
Si oui, (field in r.r.d) va être true, le || sera skippé, le true sera négativé et le preservesField sera faux alors que r.r.d[f] === r.d[f], nein ?

Reply:
oui y'a des updates comme ça. Tu sais que `false || true` ça fait true ? :D